### ISSUE TITLE
[NEW BUILD - DEFAULT / 20w17d] Build on Terrain!

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#1846]
In this build I focused on giving the player the option to build building on terrain.
To do so you put a starter block on the terrain, then you can build your building!
Once a starter block is posed, you can put any block onto the neighboring region, as the entity bounds grow by 16m on each size. I will add a way to tweak that number. Tell me if you find any issue related to that!

**Features and changes:**
- [Feature] Starter block can be build on terrain.
- [Feature] Can build any block on terrain if close to an existing terrain building (current radius of 16m).
- [Feature] Clear Item Bar row button feature.
- [Feature] Clear item from Item Bar by Shift-Click
- [Bug] Cannot click on some button for some screen ratio.
- [Bug] #1797 Collision on cylinder shape.
- [Bug] #1830 Volume Slider reset
- [Bug] #1828 Blast Door Arrow
- [Bug] #1824 Shipyard UI Tracking Cam issues
- [Bug] #1816 Raycast on small laser
- [Bug] #1814 Gravity player settings not loaded at world load
- [Bug] #1794 UI sounds not adjusting with game volume setting

=> if some kind soul could build me some nice building, for PR material or starting zone that would be cool! With this new system you will be able to import building so that will be cool to share!

![StationBuildings01](https://user-images.githubusercontent.com/6604318/80273612-ac67d800-870e-11ea-8c71-bfd8c6f9615a.jpg)

Thanks!

**Comment by ProPeach** [comment]
Can't wait to get building!
This format with all the bugs and features laid out and linked is extremely useful too, it looks very professional :stuck_out_tongue_closed_eyes: 

The concept art looks so damn cool, exciting times ahead for SEVO!

**Comment by ExodistSKY1** [comment]
Current bug that breaks building on terrain: #1852

**Comment by tsunamayo** [comment]
@ProPeach yeah, it also takes quite a bit more time to do :p... But I want to do that and post on it steam. I am just waiting for a nice screenshot of a building to show ;)
I will also store them in the game, so players knows whats up!
Cheers

**Comment by ProPeach** [comment]
Great! I sometimes think about the players who have the game but don't know about Github because they're not in the Discord server or aren't active there, it must be very confusing to them with all these changes being made especially to blueprint save locations. Putting a changelog ingame would be ideal : )
